### PR TITLE
Don't attempt to utilize local storage if its not supported

### PR DESCRIPTION
**Pull request by invisiblek**
https://github.com/AHAAAAAAA/PokemonGo-Map/issues/1907
https://github.com/AHAAAAAAA/PokemonGo-Map/issues/2121

### CODE REVIEW

**Comment by marauder37** [comment]
What platforms need to test this?

**Comment by invisiblek** [comment]
iOS safari in incognito mode apparently? The js bails without this and you get no map or no markers or something (see the linked issues above)

**Comment by kgurnani96** [comment]
just tried it, its not working.

File: static/map.js [hunk 1]
```diff
@@ -1248,3 +1260,12 @@ $(function() {
       Store.set('geoLocate', this.checked);
   });
 });
+
+function isLocalStorageSupported()  {
+  try {
+    window.localStorage.setItem("test", "1");
```
Comment (codeniko) [inline static/map.js]: Why don't you remove the item right after incase local storage is supported?

Comment (invisiblek) [inline static/map.js]: done

**Comment by kgurnani96** [comment]
@invisiblek i finally got time to check this better, first of all, you have initialised `localStorageSupported` to `true`, which actually should be the value that the function `isLocalStorageSupported()` returns. This causes that `localStorageSupported` is `true` when `initMap()` is called. I've tried setting it to false by default to know what happens, and the map doesn't even load, also `map.js:1076 TypeError: null is not an object (evaluating 'currentLocation.style')` is thrown.

**Comment by Chlodochar** [comment]
Sorry for the late review, but this doesnt make use of the default values of options at all. In addition it should save the value for the current session (locally in an object). It might be best to save values inside the Option object upon `set` and retrieve first from Option and if that's empty, retrieve and parse from localstorage, if that fails return default on `get`.

**Pull request closed without merging**